update/factory/accountDBSyncerContainerFactory.go

```diff
@@ -0,0 +1,144 @@
+}
+
+// NewAccountsDBSContainerFactory creates a factory for trie syncers container
+func NewAccountsDBSContainerFactory(args ArgsNewAccountsDBSyncersContainerFactory) (*accountDBSyncersContainerFactory, error) {
+	if check.IfNil(args.RequestHandler) {
+		return nil, update.ErrNilRequestHandler
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, update.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.TrieCacher) {
+		return nil, update.ErrNilCacher
+	}
+	if check.IfNil(args.Hasher) {
+		return nil, update.ErrNilHasher
+	}
+	if check.IfNil(args.Marshalizer) {
+		return nil, update.ErrNilMarshalizer
+	}
+	if check.IfNil(args.TrieStorageManager) {
+		return nil, update.ErrNilStorageManager
+	}
+	if args.WaitTime < time.Second {
+		return nil, update.ErrInvalidWaitTime
+	}
+
+	t := &accountDBSyncersContainerFactory{
+		shardCoordinator:   args.ShardCoordinator,
+		trieCacher:         args.TrieCacher,
+		requestHandler:     args.RequestHandler,
+		hasher:             args.Hasher,
+		marshalizer:        args.Marshalizer,
+		trieStorageManager: args.TrieStorageManager,
+		waitTime:           args.WaitTime,
+	}
+
+	return t, nil
+}
+
+// Create creates all the needed syncers and returns the container
+func (a *accountDBSyncersContainerFactory) Create() (update.AccountsDBSyncContainer, error) {
+	a.container = containers.NewAccountsDBSyncersContainer()
+
+	for i := uint32(0); i < a.shardCoordinator.NumberOfShards(); i++ {
+		err := a.createUserAccountsSyncer(i)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	err := a.createUserAccountsSyncer(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.createValidatorAccountsSyncer(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	return a.container, nil
+}
+
+func (a *accountDBSyncersContainerFactory) createUserAccountsSyncer(shardId uint32) error {
+	args := syncer.ArgsNewUserAccountsSyncer{
+		Hasher:             a.hasher,
+		Marshalizer:        a.marshalizer,
+		TrieStorageManager: a.trieStorageManager,
+		RequestHandler:     a.requestHandler,
+		WaitTime:           a.waitTime,
+		ShardId:            shardId,
+		Cacher:             a.trieCacher,
+	}
+	accountSyncer, err := syncer.NewUserAccountsSyncer(args)
+	if err != nil {
+		return err
+	}
+	trieId := genesis.CreateTrieIdentifier(shardId, state.UserAccount)
+
+	return a.container.Add(trieId, accountSyncer)
+}
+
+func (a *accountDBSyncersContainerFactory) createValidatorAccountsSyncer(shardId uint32) error {
+	args := syncer.ArgsNewValidatorAccountsSyncer{
+		Hasher:             a.hasher,
+		Marshalizer:        a.marshalizer,
+		TrieStorageManager: a.trieStorageManager,
+		RequestHandler:     a.requestHandler,
+		WaitTime:           a.waitTime,
+		ShardId:            shardId,
+		Cacher:             a.trieCacher,
+	}
+	accountSyncer, err := syncer.NewValidatorAccountsSyncer(args)
+	if err != nil {
+		return err
+	}
+	trieId := genesis.CreateTrieIdentifier(shardId, state.ValidatorAccount)
```

Comment:
I think the methods, vars, types and consts from update/genesis/base.go can be moved directly in update package. genesis package is totally unrelated.

Reply:
That is the package where the import, export keys are calculated. That is where the export and import process is done

Reply:
then the naming is wrong. Totally unrelated to "genesis" term.

Reply:
it is actually the new genesis